nexus_constructor/model/component.py

```diff
@@ -307,14 +318,16 @@ def _create_transformation_vectors_for_pixel_offsets(self) -> List[QVector3D]:
         """
         Construct a transformation (as a QVector3D) for each pixel offset
         """
-        x_offsets = self.get_field_value("x_pixel_offset")
-        y_offsets = self.get_field_value("y_pixel_offset")
-        z_offsets = self.get_field_value("z_pixel_offset")
-        if x_offsets is None or y_offsets is None:
+        try:
+            x_offsets = self.get_field_value("x_pixel_offset")
+            y_offsets = self.get_field_value("y_pixel_offset")
+        except AttributeError:
             raise Exception(
```

Comment:
Should we raise an exception here? or can we just log the error and `return`?

Reply:
Raising an exception looks right to me. The method is only used where you expect and need a result and there isn't a sensible course of action if it can't return something.